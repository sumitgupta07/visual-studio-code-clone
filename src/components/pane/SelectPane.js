import React from 'react';
import PropTypes from 'prop-types';
import { labels } from '../utils/constants';
import FileExplorer from '../menu/FileExplorer';
import Search from '../menu/Search';
import SourceControl from '../menu/SourceControl';
import Debug from '../menu/Debug';
import Extensions from '../menu/Extensions';

const SelectPane = ({ currentSelection, onSelectFile }) => {
  switch (currentSelection) {
    case labels.FileExplorer:
      return <FileExplorer onSelectFile={onSelectFile} />;
    case labels.Search:
      return <Search />;
    case labels.SourceControl:
      return <SourceControl />;
    case labels.Debug:
      return <Debug />;
    case labels.Extensions:
      return <Extensions />;
    default:
      return <FileExplorer />;
  }
};

SelectPane.propTypes = {
  currentSelection: PropTypes.string.isRequired,
  onSelectFile: PropTypes.func.isRequired
};

export default SelectPane;
