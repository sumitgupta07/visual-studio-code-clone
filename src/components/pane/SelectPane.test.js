import React from 'react';
import SelectPane from './SelectPane';
import { shallow } from 'enzyme';

let handleSelectFile;
let wrapper;
let mockProps;
beforeEach(() => {
  handleSelectFile = jest.fn();
  mockProps = {
    currentSelection: 'Search',
    onSelectFile: handleSelectFile
  };
  wrapper = shallow(<SelectPane {...mockProps} />);
});

it('should render SelectPane component', () => {
  expect(wrapper).toMatchSnapshot();
});
it('should render Search Component for currentSelection prop as Search', () => {
  expect(wrapper.find('Search').length).toEqual(1);
});

it('should not render FileExplorer component', () => {
  expect(wrapper.find('FileExplorer').length).toEqual(0);
});
