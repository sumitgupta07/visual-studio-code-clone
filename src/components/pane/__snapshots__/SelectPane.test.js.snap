// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SelectPane component 1`] = `<Search />`;
